cmp/mask/Mask.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2018 Extremely Heavy Industries Inc.
+ */
+import {Component} from 'react';
+import {PropTypes as PT} from 'prop-types';
+import {HoistComponent, elemFactory} from 'hoist/core';
+import {box} from 'hoist/layout';
+import {Classes, overlay} from 'hoist/kit/blueprint';
+
+import './Mask.scss';
+
+/**
+ * Mask for disabled or inactive components.
+ */
+@HoistComponent()
+export class Mask extends Component {
+
+    static propTypes = {
+        isDisplayed: PT.bool,
+        text: PT.string
+    };
+
+    render() {
+        let {isDisplayed, text} = this.props;
+        if (!isDisplayed) return null;
+
+        text = text ? box({cls: 'xh-mask-text', item: text}) : null;
+
+        return overlay({
+            cls: `xh-mask ${Classes.OVERLAY_SCROLL_CONTAINER}`,
+            autoFocus: false,
+            isOpen: true,
+            canEscapeKeyClose: false,
+            usePortal: false,
+            item: box({
+                cls: 'xh-mask-body',
+                alignItems: 'center',
+                justifyContent: 'center',
+                style: {
```

Comment:
Can be done away with if we don't intend to have text that would span any wider than the component we're masking. But this will make the text display in a more pleasant way if the text should need to wrap, eg for a mask on a skinny widget.